site/js/modules/webui/server.js

```diff
@@ -0,0 +1,190 @@
+        Render.makeCardObject('server-action', 'Call API', Routing.formHashPath(Routing.PATH_SERVER_CALL_API, args)),
+    ];
+
+    container.innerHTML = `
+        ${Render.cards(cards)}
+    `;
+}
+
+function handlerCallAPI(path, params, context, container) {
+    Routing.loadingStart(container);
+
+    Autograder.Server.describe()
+        .then(function(result) {
+            const endpoints = result["endpoints"];
+            const selectedEndpoint = params[Routing.PARAM_TARGET_ENDPOINT] ?? undefined;
+
+            render(endpoints, selectedEndpoint, context, container);
+        })
+        .catch(function(message) {
+            console.error(message);
+            container.innerHTML = Render.autograderError(message);
+        })
+    ;
+}
+
+function render(endpoints, selectedEndpoint, context, container) {
+    let selector = renderSelector(endpoints, selectedEndpoint);
+    let endpointArea = renderEndpointArea(endpoints, selectedEndpoint, context);
+
+    let html = `
+        <div class="page-controls">${selector}</div>
+        <div class="endpoint-area">${endpointArea}</div>
+        <div class="results-area"></div>
+    `;
+
+    container.innerHTML = html;
+
+    container.querySelector(".page-controls select").addEventListener("change", function(event) {
+        let newParams = {
+            [Routing.PARAM_TARGET_ENDPOINT]: event.target.value,
+        };
+
+        let path = Routing.formHashPath(Routing.PATH_SERVER_CALL_API, newParams);
+        Routing.redirect(path);
+    });
+
+    let button = container.querySelector(".endpoint-area button");
+
+    if (button) {
+        button.addEventListener("click", function(event) {
+            callEndpoint(selectedEndpoint, endpoints[selectedEndpoint]["input"], context, container);
+        });
+    }
+}
+
+function renderSelector(endpoints, selectedEndpoint) {
+    let optionsList = [];
+
+    for (const endpoint of Object.keys(endpoints)) {
+        let isSelected = "";
+        if (endpoint === selectedEndpoint) {
+            isSelected = "selected";
+        }
+
+        optionsList.push(`<option value="${endpoint}" ${isSelected}>${endpoint}</option>`);
+    }
+
+    return `
+        <select id="endpoint-dropdown">
+            <option value="">Select an endpoint...</option>
+            ${optionsList.join("\n")}
+        </select>
+    `;
+}
+
+function renderEndpointArea(endpoints, selectedEndpoint, context) {
+    if (!(selectedEndpoint in endpoints)) {
+        return '';
+    }
+
+    let sortedInputs = endpoints[selectedEndpoint]["input"];
+    sortedInputs.sort(function(a, b) {
+        let aPriority = fieldPriority.indexOf(a.name);
+        let bPriority = fieldPriority.indexOf(b.name);
+
+        return bPriority - aPriority;
+    });
+
+    let inputFields = [];
+    for (const field of sortedInputs) {
+        let inputType = "text";
+        let placeholder = "";
+
+        if (field.name === "user-email") {
+            placeholder = context.user.email;
```

Comment:
You can probably use the input type email for these.
https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/input/email